Documentation/network/bgp-control-plane/bgp-control-plane-v2.rst

```diff
@@ -997,13 +997,18 @@ RouterID
 --------
 
 There is ``bgpControlPlane.routerIDAllocation.mode`` Helm chart value, which stipulates how the 
-Router ID is allocated. Currently, only ``default`` is supported. In ``default`` mode,
-when Cilium runs on an IPv4 single-stack or a dual-stack, the BGP Control Plane can use the IPv4 address
-assigned to the node as the BGP Router ID because the Router ID is 32 bit-long, and we can rely on the uniqueness
-of the IPv4 address to make the Router ID unique. When running in an IPv6 single-stack, the lower 32 bits
-of MAC address of ``cilium_host`` interface are used as Router ID. If the auto assignment of
-the Router ID is not desired, the administrator needs to manually define it.
+Router ID is allocated. Currently, ``default`` and ``ip-pool`` are supported. The default allocation mode
+is ``default``.
 
+In ``default`` mode, when Cilium runs on an IPv4 single-stack or a dual-stack, the BGP Control Plane 
+can use the IPv4 address assigned to the node as the BGP Router ID because the Router ID is 32 bit-long, 
+and we can rely on the uniqueness of the IPv4 address to make the Router ID unique. When running in an IPv6 single-stack, 
+the lower 32 bits of MAC address of ``cilium_host`` interface are used as Router ID. 
+
+In ``ip-pool`` mode, users will need to provide an IPv4 IP pool to Cilium through the helm value 
+``bgpControlPlane.routerIDAllocation.ipPool``. Cilium will then assign Router IDs to BGP instances from this configured pool.
+
+If the auto assignment of the Router ID is not desired, the administrator needs to manually define it.
 In order to configure custom Router ID, you can set ``routerID`` field in an IPv4 address format.
```

Comment:
Looking at the implementation, we're not handling this case gracefully.

https://github.com/cilium/cilium/blob/038e0a74532cc9304483f0e779fa7d17ada9ee2e/operator/pkg/bgpv2/cluster.go#L385

Here we apply override, but don't check if that ID is already allocated or not. Also it doesn't record the allocated ID, so if user manually specify an ID from the pool range, ID allocator may allocate the same router ID later. I think it's better to not mention this use case until we get rid of this issue.

Reply:
Thanks for pointing it out.

The manual router ID is part of original doc https://docs.cilium.io/en/stable/network/bgp-control-plane/bgp-control-plane-v2/#bgp-configuration-override . I will push another commit to fix it and rename this PR.

Reply:
No worry. I didn't make it clear in my comments. I will fix my code and support the override. I will push another commit in this PR and rename the PR to support the override.